tests/integration/relationships/has-many-test.js

```diff
@@ -627,6 +627,43 @@ test("A hasMany relationship can be directly reloaded if it was fetched via link
   });
 });
 
+test("Calling reload multiple times does not send a new request if the first one is not settled", function(assert) {
+  assert.expect(1);
+  let done = assert.async();
+
+  Post.reopen({
+    comments: DS.hasMany('comment', { async: true })
+  });
+
+  env.adapter.findRecord = function(store, type, id) {
+    return Ember.RSVP.resolve({ id: 1, links: { comments: "/posts/1/comments" } });
+  };
+
+  let count = 0;
+  env.adapter.findHasMany = function(store, record, link, relationship) {
+    count++;
+    return new Ember.RSVP.Promise(function(resolve, reject) {
+      setTimeout(function() {
+        var value = [
+          { id: 1, body: "First" },
+          { id: 2, body: "Second" }
+        ];
+        resolve(value);
+      }, 100);
```

Comment:
I thought there must be a setTimeout in order to be "sure" the multiple reloads where not resolved. But for sure it's better if I can avoid that. Will do this :)